includes/discovery/sensors/states/hpblmos.inc.php

```diff
@@ -0,0 +1,127 @@
+<?php
+if($device['os'] == "hpblmos")
+{
+    $index = 1;
+    $state_name = 'hpblmos_FansState';
+    $state_descr = 'Fan ';
+    if ($device['hardware'] == "BladeSystem c7000 Enclosure")
+    {
+        $oids = array(
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.1',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.2',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.3',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.4',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.5',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.6',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.7',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.8',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.9',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.10',
+        );
+    } elseif ($device['hardware'] == "BladeSystem c3000 Enclosure") {
+        $oids = array(
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.1',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.2',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.3',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.4',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.5',
+           '.1.3.6.1.4.1.232.22.2.3.1.3.1.11.6',
+        );
+    }    
+
+    /* CPQRACK-MIB::cpqRackCommonEnclosureFanTable
+     * 1 - other
+     * 2 - ok 
+     * 3 - degraded
+     * 4 - failed
+     */
+
+    foreach($oids as $oid) {
+        $state = snmp_get($device, $oid, '-Oqv');
+        $descr = $state_descr . $index;
+
+        if(!empty($state))
```

Comment:
'The status could not be determined or not present.'

BTW, this would only prevent initial discovery once it has been discovered, it won't be removed.